addons/im_livechat/static/src/models/thread/thread.js

```diff
@@ -39,23 +39,55 @@ registerClassPatchModel('mail.thread', 'im_livechat/static/src/models/thread/thr
                  * of polluting the database, it is therefore acceptable and
                  * easier to handle one temporary partner per channel.
                  */
-                data2.members.push(unlink(this.env.messaging.publicPartners));
                 const partner = this.env.models['mail.partner'].create(
                     Object.assign(
                         this.env.models['mail.partner'].convertData(data.livechat_visitor),
                         { id: this.env.models['mail.partner'].getNextPublicId() }
                     )
                 );
-                data2.members.push(link(partner));
-                data2.correspondent = link(partner);
+                const publicPartnerIds = new Set(this.env.messaging.publicPartners.map(partner => partner.id));
+                this._setPartnerCommands({
+                    data: data2,
+                    predicate: (id) => publicPartnerIds.has(id),
+                    command: link(partner),
+                });
             } else {
                 const partnerData = this.env.models['mail.partner'].convertData(data.livechat_visitor);
-                data2.members.push(insert(partnerData));
-                data2.correspondent = insert(partnerData);
+                this._setPartnerCommands({
+                    data: data2,
+                    predicate: (id) => partnerData.id === id,
+                    command: insert(partnerData),
+                });
             }
         }
         return data2;
     },
+
+    //----------------------------------------------------------------------
+    // Private
+    //----------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param {Object} param0
+     * @param {Object} param0.data
+     * @param {function} param0.predicate
+     * @param {Array} param0.command
+     */
+    _setPartnerCommands({ data, predicate, command }) {
```

Comment:
This new method gave me an idea actually, instead of trying to hack the result of super, the base convert data should call itself a method "set partner", where we can just override that method, instead of trying to parse the commands. It will be less code and more robust.